### PR TITLE
EventHandlerV2 support handle event with replication.EventHeader

### PR DESCRIPTION
**Pull request by BLAZZ**
add EventHandlerV2 to get timestamp from event, so that canal can show the how much time delay when process like

```go
func (s *handler) OnXID(header *replication.EventHeader, nextPos mysql.Position) error {
	s.queue <- model.Position{
		Name:      nextPos.Name,
		Pos:       nextPos.Pos,
		Timestamp: header.Timestamp,
		Force:     false,
	}
	return nil
}
```

it can show on UI like
binlog file: mysql-bin.000102
binlog position: 1142900
**binlog timestamp: 2022-11-25 15:54:07 (1669362847)**
**tiem delay:  2 sec**

### CODE REVIEW

**Comment by atercattus** [comment]
Maybe we should change an `EventHandler` and increment semver as a major update?

**Comment by BLAZZ** [comment]
> Maybe we should change an `EventHandler` and increment semver as a major update?

change EventHandlerV2 name to another like EventHandlerWithHeader or change the method in EventHandler ?

**Comment by lance6716** [comment]
> Maybe we should change an `EventHandler` and increment semver as a major update?

Oh i forget that for major version updates we can break compatibility. I agree with this.



> > Maybe we should change an `EventHandler` and increment semver as a major update?
> 
> change EventHandlerV2 name to another like EventHandlerWithHeader or change the method in EventHandler ?

It means we can change the API interface while keeping the old name EventHandler. 

But because this chance is rare, we might ask other developers what they want to change and do them at once

File: canal/handler.go [hunk 1]
```diff
@@ -40,3 +40,24 @@ func (h *DummyEventHandler) String() string { return "DummyEventHandler" }
 func (c *Canal) SetEventHandler(h EventHandler) {
 	c.eventHandler = h
 }
+
+// EventHandlerV2 can process event with replication.EventHeader
+type EventHandlerV2 interface {
```
Comment (lance6716) [inline canal/handler.go:45]: sorry for my previous comments, now I think we can change the old `EventHandler` to this interface (will review more carefully later)

Comment (BLAZZ) [reply canal/handler.go]: I have changed EventHandlerV2 to EventHandler

**Comment by lance6716** [comment]
please fix CI, some test files need to be updated

**Review by lance6716 — APPROVED** [review]

**Pull request merged**